working_script.R

```diff
@@ -85,37 +84,37 @@ world$map$terrain[world$map$topography == "water"] <- "water"
 
 plot_topography(world) + p
 
+world <- classify_temps(world, breaks = c("polar" = -999,
+                                          "very_cold" = -35,
+                                          "cold" = -15,
+                                          "temperate" = 10,
+                                          "warm" = 25,
+                                          "very_warm" = 33))
 
 p_temp <- ggplot(data = world$map)
 p_temp <- p_temp + geom_contour(aes(x = x, y = y, z = height))
 p_temp <- p_temp + geom_contour(aes(x = x, y = y, z = as.numeric(land)), breaks = 1, linewidth = 1.5)
 p_temp <- p_temp + geom_tile(aes(x = x, y = y, fill = temperature_zone), alpha = 0.7)
 p_temp + scale_fill_brewer(palette = "RdYlBu")
 
-
-
-
 world <- gen_temperature(world,
                         pole_locs = list(c(60.5, 0)),
                         pole_radius = 5,
                         pole_power = 1,
-                        hotspot_locs = gen_hotspot_locs(world, 10),
+                        #hotspot_locs = gen_hotspot_locs(world, 10),
                         hotspot_radius = 2,
                         hotspot_power = 0.2,
+                        noise = data.frame(
+                            frequency = 0.1,
+                            amplitude = 0.1
+                        ),
```

Comment:
Could this be just a vector of length 2? That would make it easier to modify on the fly. The documentation then explains that the first value is frequency and the second is amplitude.

```suggestion
                        noise = c(0.1, 0.1),
```

Reply:
I wouldn't recommend this approach. The noise configuration is simplified at the moment because we're only using a single octave of noise with the most basic settings. Typically, you would layer multiple octaves on top of eachother to produce more variation, and have many different settings you can tweak. See https://github.com/thomasp85/ambient and https://github.com/Auburn/FastNoiseLite

Reply:
so, those would be multiple sets of frequency and amplitude?

Reply:
That would be one way of adding more complexity to the noise, however you can do all of these: https://github.com/thomasp85/ambient/blob/e561160d71da036da0b0ea189a1736e3b3c54a09/R/noise-simplex.R#L31

Reply:
got it, ok. leave as is :+1:

Reply:
Also, I just really dislike putting values that carry different meanings in the same vector (array in most languages). Good code documents itself, and it makes the meaning of the code clear when you read it back in both locations. `noise[1]` doesn't mean anything, but `noise$frequency` does.